### PR TITLE
Fixes #9388 - CNAME Support on UI

### PR DESCRIPTION
**Pull request by guimaluf**
WIP Include CNAMERecord for DNS

### CODE REVIEW

**Comment by timogoebel** [comment]
@guimaluf, thanks for your PR.

First of all, have you seen this already? https://github.com/theforeman/foreman/pull/2158/commits/7ac9b965fffe4a2a4e2d6dee4df3603996d40f55

This might give you some more ideas.

Could you briefly describe, what exactly do you want to archive with this. How do you imagine CNAME-Records to work?

**Comment by guimaluf** [comment]
@timogoebel 
Yes, I saw it! I'm using #2160  as a reference for what I'm doing.

I don't know exactly how foreman works behind the scenes, I'm still studying it.
In a high level view, I imagine two possibilities:
1. Include another input text in interface form allowing an array of alias.
2. Reuse the same input text for DNS name allowing an array of strings where the first string is the hostname and all the rest aliases.

What do you think? What detail level do you need?

I'll keep working on it and thanks for replying!

**Comment by dLobatog** [comment]
@guimaluf The hound is just being stupid, we have to fix it. Single quotes are better unless there's string interpolation, sorry for the mess of comments.

**Comment by guimaluf** [comment]
@dLobatog now I've changed all my single quotes? hahahahaha no problem, I'll rollback later

**Comment by dLobatog** [comment]
@guimaluf No worries, I don't think we enforce either style 😄 It's just that hound for some reason is using the default style they have instead of picking up our hound.yaml for some reason

**Comment by timogoebel** [comment]
> What do you think? What detail level do you need?

That's all I wanted to know. So basically you want the CNAME aliases to be deloyed when the host is created and deleted when the host is deleted, correct?

**Comment by guimaluf** [comment]
@dLobatog now I've changed all my single quotes? hahahahaha no problem, I'll rollback later!

I would like to address everyone who commented on PR #2160 and here
@timogoebel @ohadlevy @ares @dLobatog 

**How should I include host aliases on interfaces?**
Should I create a new model/table belonging to `DnsInterface`(or `Nic::Managed`? or ?) containing the aliases or a string attribute which will contain the whole list of aliases?

do you guys suggests anything better then this?

**Comment by timogoebel** [comment]
@guimaluf : Some thoughts:

- I'd create a new model (let's say `AliasRecord`). The `Nic::Base` can than have many AliasRecords.
- The `AliasRecord` should belong to a `Domain`.
- I'd suggest doing the orchestration (e.g. deploying the dns record), when the model and ui is completely done.
- You can get the smart proxy to deploy the Dns Changes from the domain.
- `DnsInterface` is just a concern. I wouldn't add the cname records here.
- I'd suggest that the UI looks similiar to Parameters. You could have a text field to specify the record name and a select field to specify the domain.

File: app/models/host_alias.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class HostAlias < ActiveRecord::Base
+  belongs_to :domains
+  belongs_to :nics
```
Comment (ohadlevy) [inline app/models/host_alias.rb]: could you add a validation that the alias is unique across the domain

Comment (guimaluf) [reply app/models/host_alias.rb]: This is WIP! :)
I'll do more stuff, just stuck with other things!

**Comment by guimaluf** [comment]
@timogoebel 
> I'd suggest that the UI looks similiar to Parameters. You could have a text field to specify the record  name and a select field to specify the domain.

Should I create a tab on Interface modal or include in the bottom the aliases fields?

**Comment by guimaluf** [comment]
screenshot from 2017-02-07 12-33-05


what do you guys think about this type of interface?

File: app/models/host_alias.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+class HostAlias < ActiveRecord::Base
+  belongs_to :domain
```
Comment (ohadlevy) [inline app/models/host_alias.rb:2]: should it be unique in the context of a domain? e.g. you can't have more than one from a domain.

Comment (guimaluf) [reply app/models/host_alias.rb:2]: I don't think so
you could have `mywebserver.example.com` been an alias to `api.foobar.com` and to `www.foobar.com`

Comment (ohadlevy) [reply app/models/host_alias.rb:2]: i mean it the other way around, only one server can have an alias for api.foobar.com

**Comment by guimaluf** [comment]
I'm struggling to write the interface/controller. I need help!

##### Can't align `remove` link on alias table with `host alias` and `domain` input
##### Can't set host_aliases parameters correctly in controller/view
host_alias input are like `host[interfaces_attributes][0][host_aliases][name]` while it should be
`host[interfaces_attributes][0][host_aliases][0][name]`. I really don't know where to set this

**Comment by ohadlevy** [comment]
@guimaluf what is the status of the PR ? is the last comment still accurate ?

**Comment by guimaluf** [comment]
@ohadlevy I needed to focus on other tasks and stop working on this.  

I'm stucked on interface process, as I mentioned above. I did not submitted my last changes, so I'm doing this right now.

I'd appreciate If anyone could help!

**Pull request closed without merging**